### PR TITLE
build: add useful workflows for maintenance

### PR DESCRIPTION
**Pull request by mariajgrimaldi**
## Description
This PR adds useful GitHub workflows for an easier maintenance job:
- Pypi publishing after new release
- Python requirements update every Monday
- Test suite for better QA

## How to test
Either the python reqs and pypi publish job are used in Open edX plugins like: 
https://github.com/openedx/tutor-contrib-coursegraph/blob/master/.github/workflows/pypi-publish.yml
A similar version: https://github.com/openedx/openedx-events/blob/main/.github/workflows/pypi-publish.yml
ad they're are proven to be working:
https://pypi.org/project/tutor-contrib-coursegraph/
Now, for the test workflow, run the following commands locally:

```
make dev-requirements
make test-lint
make test-format
make test-pythonpackage
```

## References
Coursegraph plugin existent workflows: https://github.com/openedx/tutor-contrib-coursegraph/tree/master/.github/workflows
Community health files templates: https://github.com/openedx/.github

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @mariajgrimaldi!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

File: .github/workflows/test.yml
Comment (bmtcril) [inline .github/workflows/test.yml:5]: Same thing here, no nightly branches

File: .github/workflows/test.yml
Comment (bmtcril) [inline .github/workflows/test.yml:22]: And the pinned version question

**Review by bmtcril — COMMENTED** [review]
Same feedback as the others, happy to merge all of them once that's addressed

**Comment by itsjeyd** [comment]
label: core contributor

**Review by bmtcril — APPROVED** [review]
I think this is good to go once rebased

**Pull request merged**
